apps/backend/src/api/audit.py

```
@@ -0,0 +1,197 @@
+        result = await service.create_bulk_audit_entries(bulk_data.entries)
+        return BulkAuditOperationResponse(bulk_result=result)
+    except ValidationError as e:
+        raise HTTPException(status_code=400, detail=str(e))
+    except DatabaseOperationError as e:
+        raise HTTPException(status_code=500, detail=str(e))
+
+
+@router.get("", response_model=DataCollectionAuditListResponse)
+async def list_audit_entries(
+    pagination: PaginationParams = Depends(),
+    device_ids: Optional[str] = Query(None, description="Comma-separated device UUIDs"),
```

Ruff (0.12.2)

Use `X | None` for type annotations

Convert to `X | None`

(UP045)